service/matching/taskWriter.go

```diff
@@ -216,6 +217,8 @@ func (w *taskWriter) appendTasks(
 }
 
 func (w *taskWriter) taskWriterLoop(ctx context.Context) error {
+	ctx = headers.SetCallerInfo(ctx, headers.NewCallerInfo(headers.CallerTypeBackground))
```

Comment:
Will this cause AddTask (that is not sync matched) more likely to fail?

Reply:
Double checked all matching db requests and marked all of them as background.